frontend/src/screens/apps/NewApp.tsx

```diff
@@ -171,12 +180,17 @@ const NewAppInternal = ({ capabilities }: NewAppInternalProps) => {
       throw new Error("No CSRF token");
     }
 
+    if (!permissions.scopes.size) {
+      toast({ title: "Please specify wallet permissions." });
+      return;
+    }
+
     try {
       const createAppRequest: CreateAppRequest = {
         name: appName,
         pubkey,
         budgetRenewal: permissions.budgetRenewal,
-        maxAmount: permissions.maxAmount,
+        maxAmount: permissions.maxAmount || 0,
```

Comment:
^if user doesn't change and it stays in `NaN`
This whole `NaN` thing feels hacky, but changing `maxAmount` to `number | undefined` would need a lot of changes